mitosheet/src/components/taskpanes/Graph/GraphSetupTab.tsx

```diff
@@ -0,0 +1,337 @@
+                        </p>
+                    </Col>
+                    <Col>
+                        <Select
+                            value={props.graphParams.graphCreation.graph_type}
+                            onChange={(graphType: string) => {
+                                setGraphType(graphType as GraphType)
+                            }}
+                            width='small'
+                            dropdownWidth='medium'
+                        >
+                            <DropdownItem
+                                title={GraphType.BAR}
+                            />
+                            <DropdownItem
+                                title={GraphType.LINE}
+                            />
+                            <DropdownItem
+                                title={GraphType.SCATTER}
+                            />
+                            <DropdownItem
+                                title={GraphType.HISTOGRAM}
+                            />
+                            <DropdownItem
+                                title={GraphType.DENSITY_HEATMAP}
+                            />
+                            <DropdownItem
+                                title={GraphType.DENSITY_CONTOUR}
+                            />
+                            <DropdownItem
+                                title={GraphType.BOX}
+                            />
+                            <DropdownItem
+                                title={GraphType.VIOLIN}
+                            />
+                            <DropdownItem
+                                title={GraphType.STRIP}
+                            />
+                            <DropdownItem
+                                title={GraphType.ECDF}
+                            />
+                        </Select>
+                    </Col>
+                </Row>
+
+                <AxisSection
+                    /* 
+                        We use a key here to force the Axis Section to update when the user changes the x_axis_column_ids.
+                        A key is required because react does not know that the object x_axis_column_ids changed in all cases. 
+                        Particularly, when the user changes the x_axis_column_ids from [A, B, A] to [B, A] by 
+                        deleting the first A, React does not recognize that the change has occurred and so the Axis Section does 
+                        not update even though the graph updates.
+
+                        We append the indicator xAxis to the front of the list to ensure that both AxisSections have unique keys. 
+                        When the Axis Sections don't have unique keys, its possible for the sections to become duplicated as per 
+                        the React warnings.
+                    */
+                    key={['xAxis'].concat(props.graphParams.graphCreation.x_axis_column_ids).join('')}
```

Comment:
@aarondr77 is this still true? If so, it's because we're not making good copies of things, and I expect that this can die...

Can you a) check if removing the key is working now, and b) figure out where we're updating things that we aren't making a copy of it correctly?

Not a big rush on this, but this is def a code smell, and that comment is nasty nasty. This sort of thing has led to bugs in the past, with things not updating properly (in the column control panel specifically).